### PR TITLE
Add StringLatin1.inflate unit tests

### PR DESCRIPTION
**Pull request by fjeremic**
This method will be JIT regonized via an intrinsic (work yet to come).
In anticipation of such work we add unit tests for this recognized
method for various sizes and inputs that cover most corner cases.

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by fjeremic** [comment]
@dchopra001 could you please review?

File: test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStringLatin1.java
Comment (dchopra001) [inline test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStringLatin1.java:63]: During debugging (while working on inlining `StringLatin1.inflate`), I mainly used input strings that looked like below:
```
{ '\u0052', '\u0053', '\u0054', '\u0055', '\u0056', '\u0057', '\u0058', '\u0059' };
```

If the characters are random, then it required a little more manual scanning to find where an issue is. (ex. we might miscalculate an offset and hence overwrite some data starting at an earlier point in the `actual` array)

Obviously with automated testing the character value doesn't matter. However if we run into edge case failures once the inlining code is merged, then having a predictable sequence of characters in these array could make debugging easier as it can be easier to take a quick glance at a memory and figure out where we start corrupting the data. Given that do you think it'd be better to have a more predictable sequence of characters in the input strings?

Comment (fjeremic) [reply test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangStringLatin1.java]: That's a great point. I totally agree. Fixed in [force-push](https://github.com/eclipse/openj9/compare/780a7d712a5d251f5bcfeb3b5a1451e10a1c79c0..d3df740fd2c2d6699aa531e035b481f7673e1a5c).

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8,jdk11

**Review by dchopra001 — APPROVED** [review]

**Comment by fjeremic** [comment]
@llxia do you want to be a committer for this one?

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**
